llmfoundry/data/packing.py

```diff
@@ -59,76 +100,46 @@ def efficiency(self) -> float:
     def __call__(
             self,
             examples: List[Dict[str, torch.Tensor]]) -> Dict[str, torch.Tensor]:
-        batch = self.base_collator(examples)
-
-        assert 'attention_mask' in batch
-        assert 'input_ids' in batch
-
-        for key in batch.keys():
-            assert key in [
-                'input_ids',
-                'labels',
-                'attention_mask',
-                'bidirectional_mask',
-            ]
-
         # Cut everything down to size
-        sizes, trimmed_examples = [], []
-        for idx in range(batch['attention_mask'].shape[0]):
-            size, trimmed_example = extract_trim_batch_idx(batch, idx)
-            sizes.append(size)
-            trimmed_examples.append(trimmed_example)
+        sizes = [len(example['input_ids']) for example in examples]
```

Comment:
can we assume that we no longer need to trim examples if we pack at the dataset level? 

Are datasets always unpadded?